### PR TITLE
Improvements to tabs and popup components

### PR DESCRIPTION
**Pull request by hypeJunction**
**feat(forms): anchors can now be rendered as form field type**
Adds input/anchor to allow anchors to be rendered as forms fields

**chore(js): tabs component now uses deferred object**

**feat(js): add elgg.format_element**
Adds elgg.format_element() to easily build new DOM elements.
Allows tabs component to specify the target element to load ajax data into

**feat(js): teach popup module to load ajax content**
- [ ] Update popup docs

### CODE REVIEW

File: views/default/input/anchor.php [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+/**
+ * Shortcut that allows using anchors with elgg_view_field()
+ */
+echo elgg_view('output/url', $vars);
```
Comment (hypeJunction) [inline views/default/input/anchor.php]: Doesn't really belong in `input` views. Other thoughts?

File: js/lib/elgglib.js [hunk 2]
```diff
@@ -599,3 +599,24 @@ elgg.is_in_object_array = function(object, parent, value) {
...
+/**
+ * Builds a new jQuery DOM element from an array of attributes
+ *
+ * @param {object} element Element attributes
+ *                         Optional #tag_name that defies the tag name (default: div)
+ *                         Optional #text     that defines the html content of the new element (default: '')
+ * @returns {jQuery}
+ */
+elgg.format_element = function(element) {
+	var tag_name = element['#tag_name'] || 'div';
```
Comment (hypeJunction) [inline js/lib/elgglib.js]: Should we switch to json-ld before too late? `@tag_name` both server and client side?

File: mod/developers/views/default/theme_sandbox/javascript/popup.php [hunk 3]
```diff
@@ -2,33 +2,53 @@
...
+				'my' => 'left top',
+				'at' => 'left bottom',
+			)),
+			'value' => 'Load content in a popup',
+		],
+		[
+			'#type' => 'button',
+			'class' => 'elgg-button elgg-button-submit',
+			'rel' => 'popup',
+			'data-ajax-href' => 'ajax/view/theme_sandbox/components/tabs/ajax',
```
Comment (hypeJunction) [inline mod/developers/views/default/theme_sandbox/javascript/popup.php]: Needs to be combined under single encoded `data-ajax` for both tabs and popup. Later we can add a helper AMD to do the dirty work.

File: views/default/elgg/popup.js [hunk 4]
```diff
@@ -153,15 +189,21 @@ define('elgg/popup', ['elgg', 'jquery', 'jquery-ui'], function (elgg, $) {
...
+					if ($trigger.length) {
+						$trigger.removeClass('elgg-state-active');
+						if ($trigger.data('ajaxReload')) {
+							$target.remove();
+						}
+					}
 
-				// @todo: use css transitions instead of $.fadeOut()
-				$target.fadeOut().removeClass('elgg-state-active elgg-state-popped');
+					console.log($target.trigger('close'));
```
Comment (hypeJunction) [inline views/default/elgg/popup.js]: Remove debug

**Review by hypeJunction — COMMENTED** [review]
I would like this to go into 2.3 if that's possible. Without 2.4 and undefined release date for 3.0...

**Comment by hypeJunction** [comment]
Followed by #10509

File: js/lib/elgglib.js [hunk 5]
```diff
@@ -599,3 +599,24 @@ elgg.is_in_object_array = function(object, parent, value) {
...
+	var text = element['#text'] || ''
+	
+	delete element['#tag_name'];
+	delete element['#text'];
+
+	return $('<' + tag_name + '>')
+			.addClass('hidden')
```
Comment (mrclay) [inline js/lib/elgglib.js]: Seems unnecessary

**Comment by hypeJunction** [comment]
I won't be able to work on this any time soon so ignore my comment about getting this into 2.3

**Pull request closed without merging**